nntrainer/layers/cl_layers/reshape_cl.cpp

```diff
@@ -115,185 +114,178 @@ void ReshapeLayerCl::ReshapeProcess(Tensor const &input, Tensor &output,
     const float *data = input.getData();
     float *rdata = output.getData();
     copy_cl(data, rdata, input_batch_size, input_channels, input_height,
-            input_width, context);
+            input_width);
   } else if (input.getDataType() == ml::train::TensorDim::DataType::FP16) {
 #ifdef ENABLE_FP16
     const _FP16 *data = input.getData<_FP16>();
     _FP16 *rdata = output.getData<_FP16>();
     copy_cl_fp16(data, rdata, input_batch_size, input_channels, input_height,
-                 input_width, context);
+                 input_width);
 #else
     throw std::invalid_argument("Error: enable-fp16 is not enabled");
 #endif
   }
 }
 
-void ReshapeLayerCl::copy_cl(const float *input, float *res,
-                             unsigned int input_batch_size,
-                             unsigned int input_channels,
-                             unsigned int input_height,
-                             unsigned int input_width,
-                             RunLayerContext &context) {
+void ReshapeLayerCl::copy_cl_fp16(const __fp16 *input, __fp16 *res,
+                                  unsigned int input_batch_size,
+                                  unsigned int input_channels,
+                                  unsigned int input_height,
+                                  unsigned int input_width) {
 
   bool result = false;
 
   do {
-    result = context.clCreateKernel(copy_cl_kernel_, context.LayerKernel::COPY,
-                                    ReshapeLayerCl::kernel_copy);
-    if (!result) {
+    ClContext::SharedPtrClKernel kernel_copy_ptr =
+      cl_context_ref.registerClKernel(copy_cl_kernel_fp16_, "copy_cl_fp16");
+    if (!kernel_copy_ptr) {
       break;
     }
 
```

Comment:
What about making this part as a separate function and letting the function to be called when the `clLayer` is registered? This opinion comes from #2723. Could you please think about this and share your opinion?

Reply:
About this comment, we may need additional discussions and apply it in another PR. As I understood, this commit aims to update the kernels to work with the recent commit. You can ignore this comment for this PR now :)